pkg/k8scontext/types.go

```diff
@@ -22,6 +22,7 @@ type InformerCollection struct {
 	AzureIngressProhibitedTarget cache.SharedInformer
 	IstioGateway                 cache.SharedIndexInformer
 	IstioVirtualService          cache.SharedIndexInformer
+	Nodes                        cache.SharedIndexInformer
```

Comment:
We are adding informers for Nodes only to get cluster ID.
We need to fetch the cluster ID once; Adding these informers here has long-term impact on the performance of AKS.
What if we came up with a different way to get a unique identifier for the cluster?
We could do something as simple as using the API host.